lib/libzrepl/data_conn.c

```diff
@@ -334,6 +341,12 @@ uzfs_zvol_worker(void *arg)
 				}
 			}
 
+			/* App IOs should go to cloen_zv */
+			if (!rebuild_cmd_req &&
+			    !ZVOL_IS_HEALTHY(zinfo->main_zv) &&
+			    (zinfo->clone_zv != NULL))
+				read_zv = zinfo->clone_zv;
```

Comment:
we should do the same for the sync also (ZVOL_OPCODE_SYNC) ?

Reply:
done sir.